File: Javaluator/demoSrc/examples/Restricting.java
package examples;

import net.astesana.javaluator.*;

/** An example of how to restrict operators, functions and constants of an existing evaluator.
 */
public class Restricting {
	public static void main(String[] args) {
		// Let's create a double evaluator that only support +,-,*,and / operators, with no constants, functions or parenthesis
		// First create empty evaluator parameters
		Parameters params = new Parameters();
		// Add the supported operators to these parameters
		params.add(DoubleEvaluator.PLUS);
		params.add(DoubleEvaluator.MINUS);
		params.add(DoubleEvaluator.MULTIPLY);
		params.add(DoubleEvaluator.DIVIDE);
		params.add(DoubleEvaluator.NEGATE);
		// Create the restricted evaluator
		DoubleEvaluator evaluator = new DoubleEvaluator(params);
		
		String expression = "3*-4+2";
		System.out.println (expression+" = "+evaluator.evaluate(expression));
	}
}

File: javaluator-web/en/doc/tutorial/restricting.php
<h2>Customizing built-in evaluator</h2>
<h3>Restricting an evaluator</h3>
<p>Javaluator come with some built-in functions or operators. But, for some uses, it could be too much.</p>
<p>Imagine you are developping a simple calculator to learn basic operations to children. For sure, the built-in
trigonometric functions are not valuable. The good news is that Javaluator allows you to customize built-in evaluator to recognize only
a subset of the built-in functions, operators, constants, or brackets.</p>
<p>This could be done by passing a <a href="http:javadoc/net/astesana/javaluator/Parameters.html">parameters</a> argument to the <a href="http:javadoc/net/astesana/javaluator/DoubleEvaluator.html#DoubleEvaluator(net.astesana.javaluator.Parameters)">built-in constructor</a>.</p>
<p>First, you have to create a <a href="http:javadoc/net/astesana/javaluator/Parameters.html">Parameters</a> instance, then add the functions, operators, constants and brackets you want to it.
<pre>
Parameters params = new Parameters();
// Define the parameters.
params.add(DoubleEvaluator.PLUS);
...
params.addExpressionBracket(BracketPair.PARENTHESES);
</pre>
<p>Then create the evaluator, it will only know what is defined in its parameters.</p>
<pre>
DoubleEvaluator evaluator = new DoubleEvaluator(params);
</pre>
<br/><hr/>
<p>Here is a complete working sample code:</p>
<?php displayCode($chapter);?>
<p>The output is <code>(3*-4)+2 = -10.0<br/>3^2 is an invalid expression<br/>ln(5) is an invalid expression</code></p>